syntax.cf

```diff
@@ -86,8 +86,14 @@ separator nonempty InputDefinition "" ;
 separator HiddenDefinition "" ;
 separator nonempty OutputDefinition "" ;
 
-NetworkDef. NetworkDefinition ::= "(declare-network" VariableName [InputDefinition] [HiddenDefinition] [OutputDefinition] ")" ;
--- Zero or more network definitions allowed
+-- This statement declares that a network has the same architecture as another.
+IsomorphicTo. CompStm ::= "(isomorphic-to" VariableName ")" ;
+-- This statement declares that two networks have identical architecture weights
+EqualTo. CompStm ::= "(equal-to" VariableName ")" ;
+separator CompStm "" ;
```

Comment:
So thanks to the transitivity of equality and isomorphism, I don't think we need to allow the user to provide a list of these. The user should be able to provide at most one of either `equal-to` or `isomorphic-to`.

Now exactly where in the pipeline we should enforce that I don't know. If you think it's better to allow them to provide a list in the grammar and then throw a meaningful error in the scope/type checker, then that's fine! If we do take that approach then we should document it here.